src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -235,8 +235,17 @@ IN_PROCESS_APPLICATION::ExecuteApplication()
             LOG_INFOF(L"Setting current directory to %s", this->QueryApplicationPhysicalPath().c_str());
         }
 
-        bool clrThreadExited;
+        if (m_pConfig->QueryCallStartupHook())
+        {
+            // Used to display developer exception page when there is an exception in main.
+            // TODO should we preserve whatever is in the environment variable before setting it?
+            SetEnvironmentVariable(L"DOTNET_STARTUP_HOOKS", L"Microsoft.AspNetCore.Server.IIS");
```

Comment:
Could you run the following exploratory tests and report back your findings? I want to make sure we understand the behavior of the feature:

1. What happens in the browser, on the first request, if this assembly doesn't exist (i.e. try renaming it to a bogus name)
1. What happens in the browser, on the first request, if this assembly throws from inside it's StartupHook
1. What happens in the browser, on the first request, if you have a startup error **and** you throw from within the UnhandledException handler itself.

I recognize these are super edge cases, but I want to make sure we know what's going on. If there's any way to detect these cases, we can at be prepared to diagnose issues.

Reply:
1. The app fails to start.
2. The app fails to start.
3. From what I can tell, it just causes the UnhandledException handler to exit early.